src/west/manifest.py

```diff
@@ -115,9 +115,19 @@ def __init__(self, source_file=None, source_data=None):
         Each element's values are fully initialized; there is no need
         to consult the defaults field to supply missing values.'''
 
+        self.zephyr_base = None
+        '''Absolute path to the ZEPHYR_BASE directory implied by this
+        manifest, if any. By default this is the path to the project whose
+        name is "zephyr", if one exists. The user may also override
```

Comment:
If we do switch to `env-map` then I would remove the guesswork completely and require that in the manifest.